core/lib/via_btc_client/src/indexer/parser.rs

```diff
@@ -204,12 +204,19 @@ impl MessageParser {
 
         debug!("Parsed bridge address");
 
+        // we need this to enable serde serialization and deserialization on the library structs
+        let network_unchecked_verifier_addresses: Vec<Address<NetworkUnchecked>> =
```

Comment:
Few lines above, we use `require_network()` to validate the address and convert from `NetworkUnchecked` to `NetworkChecked`, please remove this conversion instead of converting back to `NetworkUnchecked` here.

Reply:
FYI, the NetworkChecked data type doesn't implement Serde serialization and deserialization as explained by @irnb.
A solution is to extend the NetworkChecked to support Serde but at the end we will end with additional code. wdyt?

Reply:
You are right! for Address<NetworkUnchecked> it's implemented that's why it's used in this struct [SystemBootstrappingInput](https://github.com/vianetwork/via-core/blob/86b657758ad445cc4a1a8a2251726b810645be77/core/lib/via_btc_client/src/types.rs#L75)

And this is the reason why @irnb added this hack to convert the checked back to unchecked address.